src/transformStateWithClones.js

```diff
@@ -8,6 +8,51 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const ACTION_ADD = 'addProperties';
+  const ACTION_REM = 'removeProperties';
+  const ACTION_CLR = 'clear';
+  const copyState = [];
+  let obj = { ...state };
+
+  for (let i = 0; i < actions.length; ++i) {
+    const curOper = actions[i].type;
+
+    if (i > 0) {
+      obj = { ...copyState[i - 1] };
+    }
+
+    switch (curOper) {
+      case ACTION_ADD:
+        const extraData = actions[i].extraData;
+
+        for (const key in extraData) {
+          obj[key.toString()] = extraData[key.toString()];
+        }
+
+        copyState.push(obj);
```

Comment:
we can just copy the object here
```suggestion
        copyState.push({...obj});
```

Reply:
I've done it.